ros_buildfarm/config/ci_build_file.py

```diff
@@ -123,3 +123,14 @@ def __init__(self, name, data):  # noqa: D107
                     assert isinstance(plot_config_data, dict)
                     self.show_plots[plot_group].append(
                         PlotConfig(name, plot_config_data))
+
+        self.benchmark_patterns = []
+        if 'benchmark_patterns' in data:
+            self.benchmark_patterns = data['benchmark_patterns']
+            assert isinstance(self.benchmark_patterns, list)
+            assert all(isinstance(pattern, str) for pattern in self.benchmark_patterns)
+
+        self.benchmark_schema = None
+        if 'benchmark_schema' in data:
+            self.benchmark_schema = data['benchmark_schema']
```

Comment:
Could this check the Python type of the value? Maybe also check if the value is valid JSON / XML?

Reply:
6742558

Reply:
It would be nice if the error message could include the one provided by the corresponding parser to give the user enough context if the string is invalid.

Reply:
Since the assertion happens inside the error handling for the other exceptions, both of the parse exceptions will be printed.

Reply:
Well, I think from a user point of view only a single exception message should be shown. So I would suggest to first determine what format the string is then only invoke that parser.

Reply:
> So I would suggest to first determine what format the string is then only invoke that parser.

Using some heuristic? Or is there some syntactic property that I can easily check for that could rule out one format or the other?

Also, it wasn't my choice to show all 3 messages, that happens automatically. I could use some flags to avoid it, but I'm starting to think that this is getting a little involved for a simple syntax check...

Reply:
> I'm starting to think that this is getting a little involved for a simple syntax check

If the user provides a config file and it isn't valid than I think the error message should point to all the context available.

> Using some heuristic?

Maybe just checking if the string starts with a `<`? If a heuristic doesn't work for some reason you could include both parsers error message in the exception (and the user determines which one to consider).

Reply:
Here you go: 56d9537

Reply:
LGTM. Should the same check also be performed in the `source_build_file.py` file?

Reply:
Gah, you're right, I missed it. Here it is: 25e7840